<?xml version="1.0"?>

<launch>
  <arg name="joy_dev" default="/dev/input/js0" />

  <node name="joy" pkg="joy" type="joy_node" output="screen" >
    <param name="dev" value="$(arg joy_dev)" />
    <param name="autorepeat_rate" value="20" />
    <param name="deadzone" value="0.1" />
  </node>

  <!-- Note that axis IDs are those from the joystick message plus one, to be able to invert axes by specifiying either positive or negative axis numbers.-->
  <!-- Axis 2 from joy message thus has to be set as '3' or '-3'(inverted mode) below-->
  <node name="position_teleop" pkg="quad_flight" type="position_teleop" output="screen">
    <param name="x_axis" value="5"/>
    <param name="y_axis" value="4"/>
    <param name="z_axis" value="2"/>
    <param name="yaw_axis" value="1"/>
    <param name="fast_button" value="5"/>
    <param name="fast_factor" value="0.1"/>
    <param name="slow_factor" value="0.02"/>
    <param name="deadzone" value="0.1" />
  </node>
</launch>
